openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -882,34 +882,35 @@ Cache-Control: no-store
 
 When the Credential Request is invalid or unauthorized, the Credential Issuer constructs the error response as defined in this section.
 
-The following additional clarifications are provided for the following parameters already defined in section 3.1 of [@!RFC6750]:
+#### Authorization Errors {#authorization-errors}
 
-`invalid_request`:
+If the Credential Request does not contain an Access Token that enables issuance of a requested credential, the Credential Endpoint returns an authorization error response such as defined in section 3 of [@!RFC6750].
 
-- Credential Request was malformed. One or more of the parameters (i.e., `format`, `proof`) are missing or malformed.
+For the errors specific to the Credential Request such as those caused by `type`, `format`, `proof`, or encryption parameters in the request, error codes parameters defined in (#credential-request-errors) SHOULD be used instead of a generic `invalid_request` parameter defined in section 3.1 of [@!RFC6750].
 
-`invalid_token`:
+#### Credential Request Errors {#credential-request-errors}
 
-* Credential Request contains the wrong Access Token or the Access Token is missing.
+If the Credential Request is requesting for the issuance of a credential not supported by the Credential Endpoint, the HTTP response body uses the `application/json` media type with an HTTP 400 (Bad Request) status code (unless specified otherwise) and includes the following parameters with the response:
 
-The following additional error codes are specified:
+* `error`: REQUIRED. A key at the top level of a JSON object, the value of which SHOULD be a single ASCII [@!USASCII] error code from the following:
+  * `invalid_credential_request`: The Credential Request is missing a required parameter, includes an unsupported parameter or parameter value, repeats the same parameter, or is otherwise malformed.
+  * `unsupported_credential_type`: Requested credential type is not supported.
```

Comment:
`type` is specific to the `*_vc_*` (W3C) formats, right? So, probably it shouldn't be in these format-independent `error` values.

Reply:
A more general error could be `unsupported_credential`. This could be used to target non format related credential errors of this klnd.

Reply:
type is not specific to W3C formats :) it is used to refer to a claimset to be included in a credential. the definition of doctype in ISO 18013-5 is `docType is the requested document type.` and we use the following definition of doctype in VCI too
> doctype: REQUIRED. JSON string identifying the credential type as defined in [[ISO.18013-5](https://openid.github.io/OpenID4VCI/openid-4-verifiable-credential-issuance-1_0.html#ISO.18013-5)].

so I think the current text is ok

Reply:
That's a slippery slope as the spec treats any credential type information as format specific. In addition, W3C has an array of type values. I assume the issuer would like to indicate which of them is unsupported.

Reply:
how should your comment be interpreted? are you agreeing with Jacob and Pedro that a generic use of a term `type` in the spec should be revisited? i do not disagree but that should be a separate issue.